src/components/Icon/Icon.tsx

```diff
@@ -21,6 +22,29 @@ interface Props extends IconProps {}
 
 export function Icon({source, color, backdrop, accessibilityLabel}: Props) {
   const i18n = useI18n();
+  const unstableMonorail = useMonorail();
+
+  /* eslint-disable babel/camelcase */
+  useEffect(() => {
+    if (unstableMonorail) {
+      let loggedSource;
+      if (typeof source === 'function') {
+        loggedSource = source.name;
+      } else if (source === 'placeholder') {
+        loggedSource = source;
+      } else {
+        loggedSource = 'custom icon string';
+      }
+
+      unstableMonorail.produce('polaris_icons_usage/1.0', {
+        icon_source: loggedSource,
+        color,
```

Comment:
Do we care if the color, backdrop or a11y label change?

Reply:
Any of them changing triggers a re-render which I think we would want to catch. Maybe not though, what do you think @kaelig ?

Reply:
Triggering two trackings of "IconX got rendered" when what happened is its color changed from red to green seems odd to me. Taken to a silly conclusion, an icon that flashes rainbow colors would be considered the most used icon even though it was never hidden / reshown.

Reply:
- color: useful for creative direction to simulate the most recurring use cases in the masterbrand work (including dark mode)
- a11y label: this one's useful to satisfy my curiosity and see if/where we should do some a11y labelling work, or see if certain icons could use a brand new variant because a label is so misaligned with the icon

Reply:
Right, getting the values could be useful, but what Ben is saying is that if those values change then we would trigger another event for that exact same icon. Does it make sense to do this? Now that I'm thinking about it I don't think so, I'd prefer to only log when icons are initialized or when the source changes

Reply:
Where does the rerendering-with-a-different-color take place?

If it happens rarely, I'm fine with whatever solution. If it happens a lot, then it might be meaningful?

Reply:
I really don't know where it happens, but off the top of my head probably whenever an icon is interacted with (Button, Banner etc) the color or backdrop might change. I think this is going to muddy the data that we're working with so I made it log again only when the source is updated.